### PR TITLE
refactor(internal/jimmhttp/httpproxy_handler.go): defines a JIMM interface for the proxy to use.

### PR DESCRIPTION
**Pull request by alesstimec**
The httpproxy_handler was directly using jimm.JIMM, which seems a bit excessive.

Eventually i would like to un-export the jimm.JIMM type and anything that uses it should define its own interface.

## Description

The what and why - include a summary of the change, describe what it does, and include relevant motivation and context.

Fixes _JIRA/GitHub issue number_

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

File: internal/jimmhttp/httpproxy_handler.go [hunk 1]
```diff
@@ -3,23 +3,36 @@
...
+	"github.com/canonical/jimm/v3/internal/jimm/credentials"
 	"github.com/canonical/jimm/v3/internal/middleware"
+	"github.com/canonical/jimm/v3/internal/openfga"
 	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
 	"github.com/canonical/jimm/v3/internal/rpc"
 )
 
+// JIMM defines an interface used by the HTTPProxyHandler to get authentication
+// information.
+type JIMM interface {
```
Comment (ale8k) [inline internal/jimmhttp/httpproxy_handler.go:23]: Let's not do interface name shadowing and come up with a better name

File: internal/jimmhttp/httpproxy_handler.go [hunk 2]
```diff
@@ -3,23 +3,36 @@
...
+	AuthenticateBrowserSession(context.Context, http.ResponseWriter, *http.Request) (context.Context, error)
+	GetCredentialStore() credentials.CredentialStore
```
Comment (ale8k) [inline internal/jimmhttp/httpproxy_handler.go:25]: Doc the methods please.

File: internal/jimmhttp/httpproxy_handler.go [hunk 3]
```diff
@@ -3,23 +3,36 @@
...
+	GetModel(ctx context.Context, uuid string) (dbmodel.Model, error)
+	LoginWithSessionToken(ctx context.Context, sessionToken string) (*openfga.User, error)
+	UserLogin(ctx context.Context, identityName string) (*openfga.User, error)
```
Comment (ale8k) [inline internal/jimmhttp/httpproxy_handler.go:28]: You could make two interfaces here too, one  for authz/n and one called "ModelGetter"

Comment (ale8k) [reply internal/jimmhttp/httpproxy_handler.go:28]: Actually yeah, let's do this, three interfaces - Authenticator, Authoriser, ModelGetter - something like this?

Comment (SimoneDutto) [reply internal/jimmhttp/httpproxy_handler.go:28]: no please, let's not start this. We are not a 1 billion LOC project

**Pull request closed without merging**